### ISSUE TITLE
Unexpected behaviour related to the empty word with presentations

### ISSUE THREAD

**Issue by MTWhyte** [libsemigroups/libsemigroups_pybind11#52]
On the current main branch, I get the following in iPython:

```
In [1]: from libsemigroups_pybind11 import presentation, Presentation

In [2]: p = Presentation([])

In [3]: presentation.add_rule(p, [0, 1], [])

In [4]: from datetime import timedelta

In [5]: p.alphabet_from_rules()
Out[5]: <monoid presentation with 2 generators and 2 relations of length 4>

In [6]: p.validate()

In [7]: presentation.redundant_rule(p, timedelta(seconds=1))
#0: KnuthBendix: running for approx. 1000ms
#0: KnuthBendixImpl: the system is confluent already
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
Input In [7], in <cell line: 1>()
----> 1 presentation.redundant_rule(p, timedelta(seconds=1))

File ~/pybind11/libsemigroups_pybind11/libsemigroups_pybind11/presentation.py:68, in redundant_rule(p, t)
     66     return redundant_rule_strings(p, t)
     67 if isinstance(p.alphabet(), list):
---> 68     return redundant_rule_words(p, t)
     69 raise TypeError("expected a presentation over strings or lists of ints")
```

**Comment by james-d-mitchell** [comment]
This is a bug in `libsemigroups` itself, which I already fixed (inadvertently) in the `master` branch, and which will be included in the next major release. Thanks for the report!

**Comment by james-d-mitchell** [comment]
BTW: I also meant to say that you can work around this by using strings rather than lists of ints, when this issue won't come up.

**Comment by MTWhyte** [comment]
Yeah everything's working with strings. One thing that was confusing me but I have figured out is that the presentation needs to know that it is a _monoid_ presentation before this works. As in the following:

```
In [10]: p = Presentation("ab")

In [11]: presentation.add_rule(p, "ab", "")

In [12]: p
Out[12]: <semigroup presentation with 2 generators and 2 relations of length 4>

In [13]: presentation.redundant_rule(p, timedelta(seconds=1))
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
Input In [13], in <cell line: 1>()
----> 1 presentation.redundant_rule(p, timedelta(seconds=1))

File ~/pybind11/libsemigroups_pybind11/libsemigroups_pybind11/presentation.py:66, in redundant_rule(p, t)
     61 """
     62 Returns the index of the first from the end redundant rule in the
     63 presentation p.
     64 """
     65 if isinstance(p.alphabet(), str):
---> 66     return redundant_rule_strings(p, t)
     67 if isinstance(p.alphabet(), list):
     68     return redundant_rule_words(p, t)

RuntimeError: /usr/local/include/libsemigroups/present.tpp:95:validate_word: words in rules cannot be empty

In [14]: p.contains_empty_word(True)
Out[14]: <monoid presentation with 2 generators and 2 relations of length 4>

In [15]: p
Out[15]: <monoid presentation with 2 generators and 2 relations of length 4>

In [16]: presentation.redundant_rule(p, timedelta(seconds=1))
#0: KnuthBendix: running for approx. 1000ms
#0: KnuthBendixImpl: the system is confluent already
Out[16]: 2
```

This makes total sense on reflection, but I think it's non-obvious enough that I'll add this to the documentation (particularly on the member `contains_empty_word` means by "should").

**Comment by james-d-mitchell** [comment]
Resolved in v0.2.0

**Comment by james-d-mitchell** [comment]
Resolved in v0.3.0
